kong/cmd/utils/migrations.lua

```diff
@@ -175,7 +165,19 @@ end
 
 local function finish(schema_state, db, opts)
   if schema_state.needs_bootstrap then
-    log("cannot run migrations: database not bootstrapped")
+    if schema_state.legacy_invalid_state then
+      -- legacy: migration from 0.14 to 1.0 cannot be performed
+      log("cannot run migrations on a legacy %s", db.infos.db_desc)
+      return
+    end
+
+    if schema_state.legacy_is_014 then
+      log("cannot run migrations on a legacy %s; run 'kong migrations up' " ..
+          "to proceed", db.infos.db_desc)
+      return
+    end
+
+    log("cannot run migrations; run 'kong migrations bootstrap' instead")
```

Comment:
It is worth noting that those should be error logs (which will be written to stderr instead of stdout). This is why we call `error()` (instead of `log.err` + `return`). I see the below `opts.force` branch is already guilty of this